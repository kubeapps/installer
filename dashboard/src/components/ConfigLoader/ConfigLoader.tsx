import Alert from "components/js/Alert";
import React from "react";

import LoadingWrapper, {
  ILoadingWrapperProps,
} from "../../components/LoadingWrapper/LoadingWrapper";

interface IConfigLoaderProps extends ILoadingWrapperProps {
  children?: JSX.Element;
  getConfig: () => void;
  error?: Error;
}

function ConfigLoader({ getConfig, error, ...otherProps }: IConfigLoaderProps) {
  React.useEffect(() => {
    getConfig();
  }, [getConfig]);
  if (error) {
    return <Alert theme="danger">Unable to load Kubeapps configuration: {error.message}</Alert>;
  }
  return <LoadingWrapper {...otherProps} />;
}

export default ConfigLoader;
