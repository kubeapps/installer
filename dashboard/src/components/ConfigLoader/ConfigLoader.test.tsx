import Alert from "components/js/Alert";
import { mount, shallow } from "enzyme";
import context from "jest-plugin-context";

import ConfigLoader from ".";
import itBehavesLike from "../../shared/specs";

context("when the config is not ready", () => {
  itBehavesLike("aLoadingComponent", {
    component: ConfigLoader,
    props: {
      loaded: false,
      getConfig: jest.fn(),
    },
  });
});

context("when there is an error", () => {
  it("renders the error details", () => {
    const wrapper = shallow(
      <ConfigLoader error={new Error("Wrong config!")} getConfig={jest.fn()} />,
    );
    expect(wrapper.find(Alert)).toExist();
    expect(wrapper).toMatchSnapshot();
  });
});

describe("componentDidMount", () => {
  it("calls getConfig", () => {
    const getConfig = jest.fn();
    mount(<ConfigLoader getConfig={getConfig} />);
    expect(getConfig).toHaveBeenCalled();
  });
});
