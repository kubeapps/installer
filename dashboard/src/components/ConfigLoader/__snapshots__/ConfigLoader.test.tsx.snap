// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the config is not ready loading spinner matches the snapshot 1`] = `
<LoadingWrapper
  loaded={false}
/>
`;

exports[`when there is an error renders the error details 1`] = `
<Alert
  app={false}
  compact={false}
  theme="danger"
>
  Unable to load Kubeapps configuration: 
  Wrong config!
</Alert>
`;
